### ISSUE TITLE
Export Images as WYGWYS

### ISSUE THREAD

**Issue by FernandVanrie** [eLAIX/eLAIX#19]
with the current extraction of the iamges, Cropping and ochter images corrections (filters, gama etc..) are not extracted.
with the new code we take the images from the document conserving all there properties.
also the Resolution is take care off

the Resolution for the images is reprecnted by the "Hres" value

for now fixed  to 
165 (for ipad ect..
 (99 is bettter for computerscreens)
only when there are more pixels than 165 per inch the image is resized, when less we use the available pixels.

so maybe include this choise in the initiation Dialog

For testing please replace the sub ExtractWriterGraphics with
ExtractWritergraphics2 +1 aditional sub and one  extra function

sub ExtractWritergraphics2(sDestFolder)
dim odoc as object
dim oDocGraphics as object
odoc = thiscomponent
GlobalScope.BasicLibraries.loadLibrary("Tools")

oDocGraphics = odoc.GraphicObjects

Dim mFileProperties(0) As New com.sun.star.beans.PropertyValue      '***********************************
    mFileProperties(0).Name= "Hidden"                                   '\*   open a sdraw hidden document  *
    mFileProperties(0).Value=true                                   '***********************************
    Dim oDrawDoc, oDrawPage, oDrawGraphic, odesktop As Object
        oDesktop = createUnoService("com.sun.star.frame.Desktop")
    oDrawDoc = oDesktop.LoadComponentFromURL("private:factory/sdraw","_blank",0,mFileProperties())
    oDrawPage = oDrawDoc.DrawPages(0)
    oDrawGraphic = oDrawDoc.createInstance("com.sun.star.drawing.GraphicObjectShape")

dim ig as long

```
        For iG = 0 to oDocGraphics.count - 1 

 dim ographic1 as object      

            oGraphic1 = oDocGraphics(ig)' (.GetbyName(oDocGraphics.getElementNames(iG+1))
dim lh as long 
dim lW as long 
dim ct as long 
dim cb as long 
dim cl as long 
dim cr as long 
dim oh as long 
dim oW as long 
dim ph as long 
dim pW as long 


           ' xray ographic1 
lH = ographic1.height
lW = ographic1.Width
cT = ographic1.graphicCrop.Top
cB = ographic1.graphicCrop.Bottom
cL = ographic1.graphicCrop.Left
cR = ographic1.graphicCrop.Right
```

   if not isnull(ographic1.graphic) then
    if ographic1.graphic.getType = 2 then 'vector
     oH = ographic1.graphic.Size100thMM.height
     oW = ographic1.graphic.Size100thMM.Width 
    else  
    oH = ographic1.graphic.sizepixel.height/96_2540
    oW = ographic1.graphic.sizepixel.width/96_2540
    endif 
    else
    oH = LH
    oW = lW
    endif 

```
dim Hprocent as double
dim Wprocent as double  
```

   ' print oH & "-" & ct & "-" & cb 
    Hprocent =  lH/(oH-cT-cB)
    Wprocent = lW/(oW-cL-cR)

dim sGfilename as string  
sGfilename = oGraphic1.GetPropertyValue("HyperLinkName")
  dim surl as string

```
sURL = Ographic1.graphicURL 

dim oOriginalGraph as object
 oOriginalGraph = getGraphFromUrl(sURL)
dim Hres as long' xray oOriginalGraph
 Hres = int( oOriginalGraph.sizepixel.height/(ographic1.height/2540))
 if Hres < 165 then
 Hres = Hres 
 else
 Hres = 165
 endif

Dim SelSize As New com.sun.star.awt.Size
'print wprocent
SelSize.width = int(oW*Wprocent)
SelSize.height = int(oH*Hprocent)
oDrawGraphic.Size = selsize
oDrawPage.add(odrawgraphic)

oDrawPage.Width = lW
oDrawPage.Height = lH

odrawGraphic.Graphic = oOriginalGraph ' om te embedden !

Dim SelPos As New com.sun.star.awt.Point
SelPos.X = cL*Wprocent*-1
Selpos.Y = cT*Hprocent*-1
oDrawGraphic.Position = selPos
oDrawGraphic.AdjustLuminance = oGraphic1.AdjustLuminance  
```

dim oDocCtrl as object
    oDocCtrl = oDrawdoc.getCurrentController()
dim oShapeCollection as object
'**_shape must bee, slected for the Diapatcher
    oShapeCollection = createUnoService( "com.sun.star.drawing.ShapeCollection" ) 
   oShapeCollection.add( oDrawgraphic ) ' add the one shape to the collection 
   oDocCtrl.select( oShapeCollection )  
   dim odocframe as object
   dim oDispatchHelper as object 
'**_\*  
    if oGraphic1.HoriMirroredOnOddPages or oGraphic1.HoriMirroredOnEvenPages then
   oDocFrame = oDocCtrl.getFrame()
   oDispatchHelper =CreateUnoService( "com.sun.star.frame.DispatchHelper" )
   oDispatchHelper.executeDispatch( oDocFrame, ".uno:MirrorHorz", "",0, Array() ) 
   endif
    oDrawGraphic.Gamma =  oGraphic1.Gamma  
    oDrawGraphic.AdjustContrast = oGraphic1.AdjustContrast
    oDrawGraphic.AdjustRed  =  oGraphic1.AdjustRed  
    if oGraphic1.VertMirrored  then
   oDocCtrl = odrawdoc.getCurrentController()
   oDocFrame = oDocCtrl.getFrame()
   oDispatchHelper =CreateUnoService( "com.sun.star.frame.DispatchHelper" )
   oDispatchHelper.executeDispatch( oDocFrame, ".uno:MirrorVert", "",0, Array() )
endif
    oDrawGraphic.GraphicColorMode = oGraphic1.GraphicColorMode 
    oDrawGraphic.AdjustBlue = oGraphic1.AdjustBlue
'***_oGraphic1.GraphicFilter**_ comes automaticly with the graphicURL !
    oDrawGraphic.AdjustGreen =  oGraphic1.AdjustGreen  
    oDrawGraphic.Transparency  = oGraphic1.Transparency 
    pW = int(lW/100 \* Hres / 25.40)
    pH = int(lH/100 \* Hres / 25.40)
    dim afilter()
    aFilter() = array((makepropertyvalue("PixelWidth", pW))
    AddPropertyValue(aFilter(),"PixelHeight",pH)
    AddPropertyValue(aFilter(),"Compression",9) 'max
    AddPropertyValue(aFilter(),"Interlaced",0)   '1 = interlaced "

```
                ' create new name with extension ...
                dim sGraphicName as string
                sGraphicName = oGraphic1.getName() & ".png"
Export( oDrawPage, sDestFolder & sGraphicName , aFilter(), Hres )
goDlgExport.Model.lblCountGraphic.Label = ig+1
next
```

odrawdoc.dispose
end sub

function getGraphFromUrl(sFileUrl as String) as Object
  dim oprovider as object
  oProvider = createUnoService("com.sun.star.graphic.GraphicProvider") 
  Dim oPropsIN(0)as new com.sun.star.beans.PropertyValue 
  oPropsIN(0).Name  = "URL" 
  oPropsIN(0).Value = sFileUrl 
  getGraphFromUrl = oProvider.queryGraphic(oPropsIN()) 
 end function 

SUB Export( xObject, stempFileUrl As String, aFilterData )
    Dim xExporter As Object
    xExporter = createUnoService( "com.sun.star.drawing.GraphicExportFilter" )
    xExporter.SetSourceDocument( xObject )
    Dim aArgs (2) As new com.sun.star.beans.PropertyValue
    'sOriginalURL = ConvertToURL(sOriginalURL)
    aArgs(0).Name  = "FilterName"
    aArgs(0).Value = "png"
    aArgs(1).Name  = "URL"
    aArgs(1).Value = stempFileURL
    aArgs(2).Name  = "FilterData"
    aArgs(2).Value = aFilterData
    xExporter.filter( aArgs() )
END SUB

**Comment by rey-ma** [comment]
Yes, it would be really great if the properties for images set in LibreOffice were exported. That would save a lot of time going through the ILIAS editor to edit every image used.
